### PR TITLE
#472 Add python pre-commit hooks

### LINKED ISSUE #472: [FEATURE] Set up linting and mypy for python api
Add a python linter such as black or autopep8 to the husky pre-commit hooks coupled with mypy type checking.
Am update to the phony for installing hooks will be required since the above tools are dependant on python dependencies.

### PR DESCRIPTION
**Pull request by asynchroza**
How to verify:
* Make sure you've installed all the latest poetry dependencies - `poetry install` in pi-api directory
* Add some changes to a python file, for example whitelines.
* Commit the changes and the pre-commit hook should fail and remove the whitelines

resolves #472

### CODE REVIEW

File: packages/py-api/py_api/middleware/auth.py [hunk 1]
```diff
@@ -20,9 +21,9 @@ class AuthMiddleware:
     """Utility class for easily initializing all authentication middleware"""
 
     @classmethod
-    def bind(cls, app):
-        @app.middleware("http")
-        async def verify_request(request: Request, call_next):
+    def bind(cls, app: FastAPI) -> None:
+        @app.middleware("http")  # type: ignore
```
Comment (asynchroza) [inline packages/py-api/py_api/middleware/auth.py:25]: Remove ignore

File: packages/py-api/py_api/middleware/auth.py [hunk 2]
```diff
@@ -1,8 +1,9 @@
-from fastapi import Request
-from fastapi.responses import JSONResponse
+from typing import Any, Callable, Final
+
+from fastapi import FastAPI, Request
 from fastapi.encoders import jsonable_encoder
-from typing import Final
-from requests import post
+from fastapi.responses import JSONResponse
+from requests import post  # type: ignore
```
Comment (asynchroza) [inline packages/py-api/py_api/middleware/auth.py:6]: Remove ignore

**Review by IvanObreshkov — APPROVED** [review]
I tested the lining. Works as expected.

**Comment by IvanObreshkov** [comment]
@asynchroza Why you chose autopep8 over black? Just asking.

File: packages/py-api/py_api/middleware/auth.py (hunk 1 above)
Comment (asynchroza) [reply packages/py-api/py_api/middleware/auth.py]: @IvanObreshkov thank you for doing that! This was a reminded to myself.

**Comment by asynchroza** [comment]
> @asynchroza Why you chose autopep8 over black? Just asking.

@IvanObreshkov I just prefer the styling rules of autopep8 but black is also good.

**Pull request merged**